docs/dqx/docs/guide/summary_metrics.mdx

```diff
@@ -0,0 +1,339 @@
+    
+    # Trigger an action to populate metrics (e.g. count, save to a table)
+    # Metrics are only populated if an action is triggered on the DataFrame such as count or saving to a table.
+    row_count = checked_df.count()
+    
+    # Access metrics
+    metrics = observation.get
+    print(f"Input row count: {metrics['input_row_count']}")
+    print(f"Error row count: {metrics['error_row_count']}")
+    print(f"Warning row count: {metrics['warning_row_count']}")
+    print(f"Valid row count: {metrics['valid_row_count']}")
+    ```
+  </TabItem>
+</Tabs>
+
+### Writing Metrics to a Table in Batch
+
+Summary metrics can be written to a table when calling `DQEngine` methods to apply checks and write output data. When the input data is read as a batch source, metrics will be collected and written in batch.
+
+<Tabs>
+  <TabItem value="Python" label="Python" default>
+    ```python
+    from databricks.labs.dqx import check_funcs
+    from databricks.labs.dqx.engine import DQEngine
+    from databricks.labs.dqx.metrics_observer import DQMetricsObserver
+    from databricks.labs.dqx.rule import DQRowRule, DQDatasetRule, DQForEachColRule
+    from databricks.labs.dqx.config import InputConfig, OutputConfig
+    from databricks.sdk import WorkspaceClient
+
+    # Define the checks
+    checks = [
+      DQRowRule(
+        criticality="warn",
+        check_func=check_funcs.is_not_null,
+        column="col3",
+      ),
+      DQDatasetRule(
+        criticality="error",
+        check_func=check_funcs.is_unique,
+        columns=["col1", "col2"],
+      ),
+      DQRowRule(
+            name="email_invalid_format",
+            criticality="error",
+            check_func=check_funcs.regex_match,
+            column="email",
+            check_func_kwargs={"regex": "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"},
+        ),
+    ]
+
+    # Create the observer
+    observer = DQMetricsObserver(name="basic_metrics")
+
+    # Create the engine with the metrics observer
+    engine = DQEngine(WorkspaceClient(), observer=observer)
+
+    # Create the input config for a batch data source
+    input_config = InputConfig("main.demo.input_data")
+
+    # Create the output, quarantine, and metrics configs
+    output_config = OutputConfig("main.demo.valid_data")
+    quarantine_config = OutputConfig("main.demo.quarantine_data")
+    metrics_config = OutputConfig("main.demo.metrics_data")
+
+    # Read the data, apply the checks, write data to valid and quarantine tables, and write metrics to the metrics table
+    engine.apply_checks_and_save_in_table(
+        checks,
+        input_config,
+        output_config,
+        quarantine_config,
+        metrics_config
+    )
+    ```
+  </TabItem>
+</Tabs>
+
+### Writing Metrics to a Table with Streaming
+
+Summary metrics can also be written in streaming. When the input data is read as a streaming source, metrics will be written for each streaming micro-batch:
```

Comment:
Mention which methods are compatible: `apply_checks_by_metadata_and_save_in_table`, `apply_checks_and_save_in_table`. Does this work with `save_results_in_table`?